tracker/tracker/src/main/modules/network.ts

```diff
@@ -0,0 +1,313 @@
+            })
+          }
+          const method = strMethod(init.method)
+
+          const reqResInfo = sanitize({
+            url: String(input),
+            method,
+            status: r.status,
+            request: {
+              headers: reqHs,
+              body: init.body,
+            },
+            response: {
+              headers: resHs,
+              body: text,
+            },
+          })
+          if (!reqResInfo) {
+            return
+          }
+
+          app.send(
+            NetworkRequest(
+              'fetch',
+              method,
+              String(reqResInfo.url),
+              stringify(reqResInfo.request),
+              stringify(reqResInfo.response),
+              r.status,
+              startTime + getTimeOrigin(),
+              duration,
+            ),
+          )
+        })
+        .catch((e) => app.debug.error('Could not process Fetch response:', e))
+
+      return response
+    })
+  }
+  /* ====== <> ====== */
+
+  /* ====== XHR ====== */
+  const nativeOpen = XMLHttpRequest.prototype.open
+  XMLHttpRequest.prototype.open = function (initMethod, url) {
+    const xhr = this
+    setSessionTokenHeader((name, value) => xhr.setRequestHeader(name, value))
+
+    let startTime = 0
+    xhr.addEventListener('loadstart', (e) => {
+      startTime = e.timeStamp
+    })
+    xhr.addEventListener(
+      'load',
+      app.safe((e) => {
+        const { headers: reqHs, body: reqBody } = getXHRRequestDataObject(xhr)
+        const duration = startTime > 0 ? e.timeStamp - startTime : 0
+
+        const hString: string | null = ignoreHeaders ? '' : xhr.getAllResponseHeaders() // might be null (though only if no response received though)
+        const resHs = hString
+          ? hString
+              .split('\r\n')
+              .map((h) => h.split(':'))
+              .filter((entry) => !isHIgnored(entry[0]))
+              .reduce(
+                (hds, [name, value]) => ({ ...hds, [name]: value }),
+                {} as Record<string, string>,
+              )
+          : {}
+
+        const method = strMethod(initMethod)
+        const reqResInfo = sanitize({
+          url: String(url),
+          method,
+          status: xhr.status,
+          request: {
+            headers: reqHs,
+            body: reqBody,
+          },
+          response: {
+            headers: resHs,
+            body: xhr.response,
+          },
+        })
+        if (!reqResInfo) {
+          return
+        }
+
+        app.send(
+          NetworkRequest(
+            'xhr',
+            method,
+            String(reqResInfo.url),
+            stringify(reqResInfo.request),
+            stringify(reqResInfo.response),
+            xhr.status,
+            startTime + getTimeOrigin(),
+            duration,
+          ),
+        )
+      }),
+    )
+
+    //TODO: handle error (though it has no Error API nor any useful information)
+    //xhr.addEventListener('error', (e) => {})
```

Comment:
This is not an error produced by tracker, why would we need to srat' v console? Either we capture it as an error, either do nothing. But there is no usual error information  like stack etc since this one is a `ProgressEvent` - so we need to wrap it around with something. (not now)

Reply:
we can srat' v console just so we know that capturing failed